### PR TITLE
SDK parity and inbox image APIs

### PR DESCRIPTION
**Pull request by ftcvlad**
### Description of Changes

1) Drop session timeout to 23 seconds to match ios
2) Minor tweaks to annotations missed in inbox enhancements PR
3) Similarly to ObjC expose imageUrl on inbox items. For this add 2 methods: `public @Nullable URL getImageUrl(int width)` and `public @Nullable URL getImageUrl()`. Default width 300px
4) update targetSdk and compileSdk versions to 30. No [changes](https://developer.android.com/about/versions/11/behavior-changes-11) affect us.
5) version range for FirebaseMessaging [19.0.0, 22.99.99]. 19.0.0 is the 1st version supporting androidx. 22.0.0 is currently latest version of FM. Token retrieval/deletion API changed along the way, so, support 2 API versions. 

Range [19.0.0, 22.99.99] means that:
- If there is only 1 dependency on FM, latest version from the range is selected. 
- If there is a specific version in the range, it is selected
- If there is a specific version exceeding the range, outcome depends on the build tool: with [gradle](https://docs.gradle.org/current/userguide/dependency_resolution.html#sub:resolution-strategy) specific version is selected, with [maven](https://maven.apache.org/pom.html#dependency-version-requirement-specification) build fails. 

6) Thread safety of InAppMessagePresenter (#61)

### Breaking Changes

- None

### Release Checklist

Prepare:

- [x] Detail any breaking changes. Breaking changes require a new major version number
- [x] Check `./gradlew assemble` passes w/ no errors

Bump versions in:

- [x] README.md

Release:

- [ ] Squash and merge to master
- [ ] Delete branch once merged
- [ ] Create tag from master matching chosen version
- [ ] Verify & Publish release from [OSS Nexus UI](https://oss.sonatype.org/#stagingRepositories)

### CODE REVIEW

File: kumulos/src/main/java/com/kumulos/android/KumulosInApp.java [hunk 1]
```diff
@@ -117,7 +117,7 @@ public static void setOnInboxUpdated(@Nullable InAppInboxUpdatedHandler inboxUpd
      * @param context
      * @param inboxSummaryHandler handler
      */
-    public static void getInboxSummaryAsync(Context context, @Nullable InAppInboxSummaryHandler inboxSummaryHandler) {
+    public static void getInboxSummaryAsync(Context context, InAppInboxSummaryHandler inboxSummaryHandler) {
```
Comment (cgwyllie) [inline kumulos/src/main/java/com/kumulos/android/KumulosInApp.java:120]: Why removing Nullability annotation? (should it be @nonnull)

File: kumulos/src/main/java/com/kumulos/android/MediaHelper.java [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import androidx.annotation.NonNull;
+
+import java.net.MalformedURLException;
+import java.net.URL;
+
+public class MediaHelper {
+
+    private static final String MEDIA_RESIZER_BASE_URL = "https://i.app.delivery";
+
+    static @NonNull URL getCompletePictureUrl(String pictureUrl, int width) throws MalformedURLException{
```
Comment (cgwyllie) [inline kumulos/src/main/java/com/kumulos/android/MediaHelper.java:12]: Specify String arg as @nonnull too?

File: kumulos/src/main/java/com/kumulos/android/KumulosInApp.java (hunk 1 above)
Comment (ftcvlad) [reply kumulos/src/main/java/com/kumulos/android/KumulosInApp.java:120]: it cannot be null (i think it was changed during implementation, but annotation stayed). Could add NonNull, but we normally assume it's required even without annotation

Comment (cgwyllie) [reply kumulos/src/main/java/com/kumulos/android/KumulosInApp.java:120]: True, but annotations hint/help devs / tooling will pick up potential NPE. Happy either way, seems it was previously incorrect.

Adding @NonNull for both args would be best I suppose, but not essential to release this I think.

File: kumulos/src/main/java/com/kumulos/android/MediaHelper.java (hunk 2 above)
Comment (ftcvlad) [reply kumulos/src/main/java/com/kumulos/android/MediaHelper.java]: y

File: kumulos/src/main/java/com/kumulos/android/KumulosInApp.java (hunk 1 above)
Comment (ftcvlad) [reply kumulos/src/main/java/com/kumulos/android/KumulosInApp.java]: updated

**Pull request merged**